compiler/modules/types/list.bal

```diff
@@ -310,7 +306,8 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, ListC
         // SemType d1 = diff(s[1], t[1]);
         // return !isEmpty(cx, d1) &&  tupleInhabited(cx, [s[0], d1], neg.rest);
         // We can generalize this to tuples of arbitrary length.
-        foreach int i in 0 ..< int:max(members.initial.length(), neg.maxInitialLen) {
+        int maxInitialLen = int:max(members.initial.length(), neg.maxInitialLen);
```

Comment:
The comment above `now we have nt.members.length() <= len` is no longer correct.